### ISSUE TITLE
[0.85.0] Failed to install package from on-prem builder

### ISSUE THREAD

**Issue by jsirex** [habitat-sh/habitat#6894]
Run the command: `RUST_LOG=debug hab pkg install tai/pre-commit --url https://bldr.mycorp.local`

```
...
↓ Downloading ya/pre-commit/1.17.0/20190823085024
[2019-08-23T10:37:38Z DEBUG habitat_http_client::api_client] GET https://bldr.mycorp.local/v1/depot/pkgs/ya/pre-commit/1.17.0/20190823085024/download?target=x86_64-linux with ApiClient { endpoint: "https://bldr.mycorp.local/v1", inner: Client }
[2019-08-23T10:37:38Z DEBUG habitat_api_client::builder] Downloading file to path: /home/sirex/.hab/cache/artifacts
[2019-08-23T10:37:38Z DEBUG hyper::client::connect::dns] resolving host="bldr.mycorp.local"
[2019-08-23T10:37:38Z DEBUG hyper::client::connect::http] connecting to 10.6.125.15:443
[2019-08-23T10:37:38Z DEBUG tokio_reactor] adding I/O source: 20971520
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Write for: 0
[2019-08-23T10:37:38Z DEBUG hyper::client::connect::http] connected to Some(V4(10.6.125.15:443))
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG hyper::proto::h1::io] flushed 231 bytes
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG tokio_reactor::registration] scheduling Read for: 0
[2019-08-23T10:37:38Z DEBUG hyper::proto::h1::io] read 8192 bytes
[2019-08-23T10:37:38Z DEBUG hyper::proto::h1::io] parsed 14 headers
[2019-08-23T10:37:38Z DEBUG hyper::proto::h1::conn] incoming body is chunked encoding
[2019-08-23T10:37:38Z DEBUG hyper::proto::h1::decode] incoming chunked header: 0x1EAB (7851 bytes)
[2019-08-23T10:37:38Z DEBUG reqwest::async_impl::response] Response: '200 OK' for https://bldr.mycorp.local/v1/depot/pkgs/ya/pre-commit/1.17.0/20190823085024/download?target=x86_64-linux
[2019-08-23T10:37:38Z DEBUG habitat_api_client::response] Response Status: 200
[2019-08-23T10:37:38Z DEBUG tokio_reactor] dropping I/O source: 0
✗✗✗
✗✗✗ We tried 5 times but could not download ya/pre-commit/1.17.0/20190823085024. Giving up.
✗✗✗
```

I believe it can be easily reproduced by point `hab` to the url of `builder-api` directly: `--url http://builder-api-server:9636`

The only differences I see - bldr.habitat.sh uses HTTP/2 (and content-length header in response), but my on-prem bulder is using HTTP/1.1 with chunked content.

I also see changes in git related to `pkg install`. Probably this is connected.

Now I'm implementing http2 support for my builder. Want to see will it fix the issue.

**Comment by jsirex** [comment]
After digging in I've found that possible issue with "content-length" response header.
If I hardly set it in nginx like: `add_header "content-length" 10000000` I able to install package.

**Comment by jsirex** [comment]
My current workaround that works:

```nginx
    location ~* ^/v1/depot/.*/download$ {
      add_header Cache-Control "private, no-cache, no-store";
      add_header content-length 128;
      proxy_pass http://backend;
    }
```

It breaks progress bar however.

**Comment by mwrock** [comment]
This should be resolved now by #6891
